### PR TITLE
test: Implement unit tests for CWallet::CreateTransaction

### PR DESCRIPTION
**Pull request by xdustinface**
I decided to add this because i worked on a fix for `CreateTransaction` and while doing that i figured its very hard to say for sure by just manual test/reviewing if things are still good since this is quite complex method imo with a lot possible routes. I think this should catch all currently possible cases in there. If someone can think about any improvement or more test cases, let me know!

Requires #3666 and #3668 to pass all the tests.

### CODE REVIEW

**Comment by xdustinface** [comment]
Ready for review!

**Comment by PastaPastaPasta** [comment]
You should probably move #3668 into this PR, cause unit tests need to be passing for us to merge this

**Comment by xdustinface** [comment]
> Requires #3666 and #3668 to pass all the tests.

Hmm imo separating them makes sense (depending on what will happen during review process :D). Why not just merge #3666 and #3668 first and then rebase this one as the others still pass without this PR. You can still verify #3668 locally with this PR for now. Still prefer a merge? Let me know!

**Comment by UdjinM6** [comment]
I'd prefer rebasing this on top of develop after 3668 is merged.

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/wallet/test/wallet_tests.cpp [hunk 1]
```diff
@@ -697,4 +699,449 @@ BOOST_FIXTURE_TEST_CASE(ListCoins, ListCoinsTestingSetup)
...
+        BOOST_CHECK(fChangeTestPassed);
+        if (!fChangeTestPassed) {
+            return false;
+        }
+        // Verify the change is at the requested position if there was a request
+        if (nChangePosRequest != -1 && !CheckEqual(nChangePosRequest, nChangePos)) {
+            return false;
+        }
+        // Verify the number of requested outputs does match the resulting outputs
+        CheckEqual(vecEntries.size(), wtx.tx->vout.size() - (nChangePos != -1 ? 1 : 0));
```
[the remark this answers is not in the document]
Comment (xdustinface) [reply src/wallet/test/wallet_tests.cpp]: What about 29b9e0f?

File: src/wallet/test/wallet_tests.cpp [hunk 2]
```diff
@@ -697,4 +699,449 @@ BOOST_FIXTURE_TEST_CASE(ListCoins, ListCoinsTestingSetup)
...
+BOOST_FIXTURE_TEST_CASE(CreateTransactionTest, CreateTransactionTestSetup)
+{
+    minRelayTxFee = CFeeRate(DEFAULT_MIN_RELAY_TX_FEE);
+    // Try to send dust
+    BOOST_CHECK(CreateTransaction({{10, false}}, false));
+    BOOST_CHECK(CreateTransaction({{10, true}}, false));
+    BOOST_CHECK(CreateTransaction({{100, false}}, false));
+    BOOST_CHECK(CreateTransaction({{100, true}}, false));
+    BOOST_CHECK(CreateTransaction({{10000, true}, {100, true}}, false));
+    BOOST_CHECK(CreateTransaction({{10000, false}, {100, false}}, false));
```
[the remark this answers is not in the document]
Comment (xdustinface) [reply src/wallet/test/wallet_tests.cpp]: Hm i just dropped those initial tests, c7e34f5. They are anyway covered in the rest of the test cases.

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**